guidelines/backend-coding-standard/backend-coding-standard.md

```diff
@@ -0,0 +1,40 @@
+# Backend Coding Standard Draft
+
+## Coding style
+- function should be stateless, pure
+- function should not have side effect, i.e. interact with the environment outside the function, causing mutatation with other things that is not expected to change
+- input id instead of the object itself to the function, then query by id inside the function
+- function name should be describing the business action instead of implementation detail (except for private functions)
+  - e.g. prefer `consume_discount_code` to `decrease_discount_code_quota_by_one`
+- be caution of n + 1 query
+- consider usage of in memory or query from db
+  - e.g. `select` / `where`
+
+## Modules
+- idea: consider `thin model, thin controller, fat service / modules`
+- only 入口 for controllers
+- encapsulation of business actions
+- include `singleton`
+- command only, query no need to be here
+  - command query separation, grouping commands with similiar business domain together for easy changing business behavior
+- grouping with domain
+  - `fulfill_order` and `create_order` should be under `order` domain 
```

Comment:
In fact, at this moment, we are more on grouping with aggregate root than domain.